tutorial/nixify/1_nix_flake.md

```diff
@@ -0,0 +1,223 @@
+        program = "${pkgs.hello}/bin/hello";
+      };
+      devShells.${system}.default = pkgs.mkShell {
+        buildInputs = [
+          pkgs.hello
+        ];
+      };
+    };
+}
+```
+In the upcoming subsections we will break down each part of the code block above.
+
+### Inputs [^1]
+
+[^1]: There are two ways to access the attributes of `inputs` within `outputs`:
+      - Adding the attribute as a parameter to `outputs`, like `outputs = { self, <attribute> }`. This allows you to use the `<attribute>` without requiring any prefix. 
+      - Bind a variable to all the parameters of `outputs`, like `outputs = inputs@{self, ...}`. This enables you to access any attribute from `inputs` in this fashion: `inputs.<attribute>`.
+[^2]: `nixpkgs-unstable` branch is named as such because of the frequent updates it receives and doesn't imply that it is unsafe.
+[^3]: `self` refers to the final state of attributes in the `outputs`. For example, `self.packages.${system}.default` refers to the attribute after assigning `pkgs.hello` to it.
+
+Nix Flake uses two different ways to specify the location of an input flake, attribute set representation and URL-like syntax. In this post we will be using the latter for the sake of simplicity, to read more about these representations [refer here](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake.html#url-like-syntax).
+
+For this example we will use [GNU hello](https://www.gnu.org/software/hello) package from `nixpkgs-unstable`[^2] branch of [`nixpkgs`](https://github.com/NixOS/nixpkgs) repository. 
+
+### Outputs
+
+- A function that accepts a required parameter `self`[^3] and optionally attributes from `inputs` as parameters. [Refer here](https://nixos.wiki/wiki/Flakes#Output_schema) for a detailed schema of `outputs`. 
+- We will include `nixpkgs` as parameter to use its attributes to define our attributes in the sections below.
+- In the `let` block we also define two things, `system` ("aarch64-darwin" here) and `pkgs` consits of all the packages for `system`. In our example `system` is hardcoded for one system, [forAllSystems](https://zero-to-nix.com/concepts/flakes#system-specificity) can be used to define packages for an array of systems.
+
+#### Packages
+
+- Attributes in `packages.${system}` points to [derivation](https://nixos.org/manual/nix/stable/language/derivations.html) that can be used to build the package.
+- `nix build` will build the derivation of the package with the name `default`. Run `nix build .#<packageName>` to build a pacakge with a specific name.
+
+#### Apps
+
+- `apps.${system}.<appName>` refers to an attrset with two required attributes `type` and `program`. The `type` attribute determines how the program should be executed, For example, "shell" for a shell script, "python" for a Python script, or "app" for an executable. On the other hand, the `program` attribute is a string that represents the path in the Nix store where the executable is located.
```

Comment:
```suggestion
- `apps.${system}.<appName>` refers to a flake app that can be run using `nix run`. It is attrset containing two keys `type` and `program`. The `type` attribute determines how the program should be executed, For example, "shell" for a shell script, "python" for a Python script, or "app" for an executable. On the other hand, the `program` attribute is a string that represents the path in the Nix store where the executable is located.
```

Reply:
> The `type` attribute determines how the program should be executed, For example, "shell" for a shell script, "python" for a Python script, or "app" for an executable

Can you link to the docs for `type`?

Reply:
I couldn't find the docs for this, I got the info from the source code.